tools/harness-thci/OpenThread.py

```diff
@@ -1389,8 +1395,11 @@ def setDefaultValues(self):
         # indicate that the channel has been set, in case the channel was set
         # to default when joining network
         self.hasSetChannel = False
-        # indicate whether the default domain prefix is used.
-        self.__useDefaultDomainPrefix = True
+        if self.DeviceCapability == OT12BR_CAPBS:
+            # indicate whether the default domain prefix is used.
+            self.__useDefaultDomainPrefix = True
+        else:
+            self.__useDefaultDomainPrefix = False
```

Comment:
```suggestion
        # indicate whether the default domain prefix is used.
        self.__useDefaultDomainPrefix = (self.DeviceCapability == OT12BR_CAPBS)
```
nit

Reply:
The issue we encountered was that the OMR prefix conflicts with the DUA prefix.  Since you can only have 1 OMR prefix, once the DUA prefix is configured and an ip address instantiated, you can't allocate an ip address on the OMR prefix.  These changes were to address that issue.  We saw this issue on the reference devices in tests DH_3_1 and DH_3_2.  With these change, those test cases could pass against our soc DUT because the OMR prefix that the TH configured was allowed on the network.  I suspect this change would cause those cases to fail once again.